lib/puppet/module.rb

```diff
@@ -128,6 +128,8 @@ def has_metadata?
       end
 
       return nil unless Puppet::FileSystem.exist?(full_path)
+      # TODO: line left in while testing to see if this is necessary
+      Puppet.warning("Unexpanded path #{full_path}") if full_path.upcase != Puppet::FileSystem.expanded_path_string(full_path).upcase
```

Comment:
That note was more for me while running through tests.  In prior iterations of this PR I returned `expanded_path_string`.  But when I moved to just setting the path to expanded in the initializer, I presumed I could remove all the other calls.  This is just a sanity check in that vein.